### PR TITLE
Add the Tetra Legends Tetra-X randomizer

### PR DESCRIPTION
**Pull request by nathyong**
This randomizer is a slightly cleaned up version of Dr Ocelot's
JavaScript variant[1], including the 1000-piece selection mechanism.

[1]: https://github.com/Dr-Ocelot/tetralegends/blob/8d57a703224cc5aa5260e13160e973755c7551d9/script/game/modules/randomizers.js

### CODE REVIEW

File: tetris/randomizers/tetra.lua [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        end
+    end
+
+    -- piece has not been generated in the last 13+
+    for piece, val in pairs(self.lastseen) do
+        if val >= 13 then
+            generated = piece
+            break
+        end
+    end
```
Comment (nathyong) [inline tetris/randomizers/tetra.lua:43]: This code flow is copied from the JS implementation, but an easy optimization here would be to invert this logic and avoid calling the RNG if a piece needs to be forced.

File: tetris/randomizers/tetra.lua [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    self.count[generated] = self.count[generated] + 1
+    self.totalcount = self.totalcount + 1
+
+    -- shuffle the piece selection for the next time
+    self.pieceselection = {}
+    for piece, count in pairs(self.count) do
+        local probability = (self.totalcount - count) / (self.totalcount * 6)
```
Comment (nathyong) [inline tetris/randomizers/tetra.lua:61]: Should probably document why magic number `6` is used here? (it's #piece_types - 1)

Comment (nathyong) [reply tetris/randomizers/tetra.lua:62]: Implemented in a4c6509

**Comment by SashLilac** [comment]
Thank you very much! Merged.

**Pull request merged**